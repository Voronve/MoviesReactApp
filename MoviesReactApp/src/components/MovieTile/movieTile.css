.movieTile {
    position: relative;
    color: #8b8888;
    width: 322px;
    cursor: pointer;
    margin-bottom: 40px;
}

.movieTile .movieInfo .flex-container {
    display: flex;
    justify-content: space-between;
}

.movieTile .movieInfo .ganres {
    text-align: start;
}

.movieTile .movieInfo .year {
    border: 1px solid #8b8888;
    border-radius: 3px;
    padding: 2px 7px;
}

.movieTile .editMenuWrapper {
    font-size: 30px;
    color: Gray;
    display: inline-block;
    position: absolute;
    top: 10px;
    right: 10px;
}

.movieTile .editMovieMenu {
    margin: 0;
    position: absolute;
    top: 0;
    right: 0;
    background-color: #424242;
    cursor: pointer;
    list-style: none;
    display: none;
    border-radius: 3px;
    padding-left: 0;
}

.movieTile .editMovieMenu.active {
    display: block;
}

.movieTile .editMovieMenu li {
    font-size: 20px;
    padding: 10px 20px;
}

.movieTile .editMovieMenu li.close {
    text-align: end;
    padding: 0 14px 0 0;
}
.movieTile .editMovieMenu li:hover {
    background-color: red;
    color: white;
}