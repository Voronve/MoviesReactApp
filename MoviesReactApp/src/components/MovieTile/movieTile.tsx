import './movieTile.css';
import { Link, useSearchParams } from 'react-router-dom';
import formQueryParamString from '../../helpers/formQueryParamsString';
export interface movieTileProps {
    /** Movie tile data object*/
    movieData: {
        /**Movie id */
        id: string,
        /** Image url*/
        poster_path: string,
        /**Movie title */
        title: string,
        /**Movie release year */
        release_date: number,
        /**Movie janres list */
        genres: string[]
    }
}

/** Single movie tile*/
export function MovieTile({ movieData }: movieTileProps ) {
    const {id, poster_path, title, release_date, genres} = movieData;
    const [searchParams] = useSearchParams();
    const resultString = formQueryParamString(searchParams);

    return (
        <Link to={`/${id}${resultString}`}>
            <div className="movieTile">
                <img src={poster_path} alt={title} width="322px"/>
                <div className="movieInfo">
                    <div className="flex-container">
                        <div className="name">{title}</div>
                        <div className="year">{`${release_date}`.split('-')[0]}</div>
                    </div>
                    <div className="ganres">{genres.join()}</div>
                </div>
            </div>
        </Link>
    );
}