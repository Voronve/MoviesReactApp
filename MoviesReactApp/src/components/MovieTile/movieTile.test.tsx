import { MovieTile } from './movieTile';
import { fireEvent, render, screen } from '@testing-library/react';

describe("Testing MovieTile component", () => {
    const onClick = jest.fn();
    const movieDataMock = {
        poster_path: "Comedy/Barbie.jpg",
        title: "Barbie",
        release_date: 2023,
        genres: ["COMEDY", "HORROR"],
        onClick
    }

    test("Testing movie tile rendering", () => {

        const { asFragment } = render(<MovieTile movieData={movieDataMock} />);

        expect(asFragment()).toMatchSnapshot();
    });

    test("Testing that callback function is working as expected", () => {
        render(<MovieTile movieData={movieDataMock} />)

        fireEvent.click(screen.getByText(movieDataMock.title));

        expect(onClick).toBeCalledWith(movieDataMock.title);
    });
});