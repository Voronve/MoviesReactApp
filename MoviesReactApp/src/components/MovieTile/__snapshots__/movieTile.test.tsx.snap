// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieTile component Testing movie tile rendering 1`] = `
<DocumentFragment>
  <a
    href="/123"
  >
    <div
      class="movieTile"
    >
      <img
        alt="Barbie"
        src="Comedy/Barbie.jpg"
        width="322px"
      />
      <div
        class="movieInfo"
      >
        <div
          class="flex-container"
        >
          <div
            class="name"
          >
            Barbie
          </div>
          <div
            class="year"
          >
            2023
          </div>
        </div>
        <div
          class="ganres"
        >
          COMEDY,HORROR
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
